Initial proposal for ICMP forwarding and ECN for flow mode (#6)
mirjak/draft-kuehlewind-masque-connect-ip: draft-kuehlewind-masque-connect-ip.md

## ECN {#ECN}

ECN requires coordination with the end-to-end communication points as
it should only be used if the endpoints are also capable and willing
to signal congestion notifications to the other end and react
accordingly if a congestion notification is received.

> The probing and verification in the ULP of end-to-end ECN requires per

Reviewer: ULP?

Reviewer 2: Upper Layer Protocol.

Author: ```suggestion
The probing and verification in the upper layer protocol of end-to-end ECN requires per
```

  setting and reading.

  b) use 4 different context IDs depending on what ECN field value was
  received or should be set.

This is work in process and will be further specified in a future
version of this document.

## ICMP handling {#ICMP}

ICMP messages are directly forwarded in tunneling mode. In flow
forwarding mode a ICMP datagram format ({{ICMP-message-format}}) is
used to send the information from some ICMP message to the client. 

The proxy upon receiving an ICMP message with a destination of an IP
address it performs flow forwarding on it needs to process the ICMP
message. First it should validate that the ICMP message and find if it
matches any of its IP flow selectors (including Conn-ID). In case
there are multiple matching use the IP selector with the most number
of field that matches fully.

Some messages may be applicable both to the proxy and the client. For
example an verified ICMPv6 Packet Too Big is applicable both to the
proxy and the client. Others like ICMPv6 Destiantion Unreachable
(Type=1), Code=3 (Address unreachable) and Code=4 (Port unreachable)
is only possible to act on by the client.

> QUESTION: Any ICMP messages that should be suppressed by the proxy?

Reviewer 2: Proxy should not suppress the ICMP messages unless -
- the ICMP messages is meant to be consumed by the proxy
- the ICMP messages is not meant to be forwarded to the particular client.

Author: ```suggestion
QUESTION: Which ICMP messages should be suppressed by the proxy?
```
